src/objects/p5.Color.js

```diff
@@ -16,27 +16,31 @@ define(function(require) {
    */
   p5.Color = function (pInst, vals) {
     this.color_array = p5.Color._getFormattedColor.apply(pInst, vals);
-    this._normalizeColorArray(pInst);
+    this._converted_color = this._convertTo255(pInst);
+    
+    var isHSB = pInst._colorMode === constants.HSB;
 
-    if (pInst._colorMode === constants.HSB) {
+    if (isHSB) {
       this.hsba = this.color_array;
-      this.rgba = color_utils.hsbaToRGBA(this.hsba);
+      this.rgba = color_utils.hsbaToRGBA(this._converted_color);
     } else {
-      this.rgba = this.color_array;
+      this.rgba = this._converted_color;
       this.hsba = color_utils.rgbaToHSBA(this.rgba);
     }
 
     return this;
   };
 
-  p5.Color.prototype._normalizeColorArray = function (pInst) {
+  p5.Color.prototype._convertTo255 = function (pInst) {
     var isRGB = pInst._colorMode === constants.RGB;
     var maxArr = isRGB ? pInst._maxRGB : pInst._maxHSB;
-    var arr = this.color_array;
-    arr[0] *= 255 / maxArr[0];
-    arr[1] *= 255 / maxArr[1];
-    arr[2] *= 255 / maxArr[2];
-    arr[3] *= 255 / maxArr[3];
+    // var arr = this.color_array;
+    var arr = [];
+    arr[0] = this.color_array[0] * 255 / maxArr[0];
+    arr[1] = this.color_array[1] * 255 / maxArr[1];
+    arr[2] = this.color_array[2] * 255 / maxArr[2];
+    arr[3] = this.color_array[3] * 255 / maxArr[3];
```

Comment:
Given that we need to index into `maxArr` regardless, there's diminishing returns to this change; I'm just used to avoiding array assignment by index (`.push()` instead of `[x] =`, that is to say)